contracts/service-registry/src/contract.rs

```diff
@@ -257,6 +258,12 @@ pub mod execute {
 
         for chain in chains {
             WORKERS_PER_CHAIN.save(deps.storage, (&service_name, &chain, &info.sender), &())?;
+
+            let existing_chains =
+                CHAINS_PER_WORKER.may_load(deps.storage, (&service_name, &info.sender))?;
+            let mut chains_names = existing_chains.unwrap_or_else(HashSet::new);
+            chains_names.insert(chain.clone());
+            CHAINS_PER_WORKER.save(deps.storage, (&service_name, &info.sender), &(chains_names))?;
```

Comment:
this operation is incredibly expensive. Currently, for each new chain, all the chains of the worker are loaded, one chain is added and the whole thing is saved back. Split it up:
- load the stored value
- add all the chains
- save it back